packages/auth/src/grant/model.ts

```diff
@@ -61,6 +66,70 @@ export class Grant extends BaseModel {
   public interactNonce?: string // AS-generated nonce for post-interaction hash
 }
 
+interface ToOpenPaymentsInteractiveGrantArgs {
+  authServerUrl: string
+  client: {
+    name: string
+    uri: string
+  }
+  waitTimeSeconds?: number
+}
+
+export function toOpenPaymentsInteractiveGrant(
+  grant: Grant,
+  args: ToOpenPaymentsInteractiveGrantArgs
+): OpenPaymentsInteractiveGrant {
+  if (!isInteractiveGrant(grant)) {
+    throw new Error('Expected interactive grant')
+  }
+
+  const { authServerUrl, client, waitTimeSeconds } = args
+
+  const redirectUri = new URL(
+    authServerUrl + `/interact/${grant.interactId}/${grant.interactNonce}`
+  )
+
+  redirectUri.searchParams.set('clientName', client.name)
+  redirectUri.searchParams.set('clientUri', client.uri)
+
+  return {
+    interact: {
+      redirect: redirectUri.toString(),
+      finish: grant.interactNonce
+    },
+    continue: {
+      access_token: {
+        value: grant.continueToken
+      },
+      uri: `${authServerUrl}/auth/continue/${grant.continueId}`,
+      wait: waitTimeSeconds
+    }
+  }
+}
+
+interface ToOpenPaymentsGrantArgs {
+  authServerUrl: string
+}
+
+export function toOpenPaymentsGrant(
+  grant: Grant,
+  args: ToOpenPaymentsGrantArgs,
+  accessToken: AccessToken,
+  accessItems: Access[]
+): OpenPaymentsGrant {
+  return {
+    access_token: toOpenPaymentsAccessToken(accessToken, accessItems, {
+      authServerUrl: args.authServerUrl
+    }),
+    continue: {
+      access_token: {
+        value: grant.continueToken
+      },
+      uri: `${args.authServerUrl}/continue/${grant.continueId}`
+    }
+  }
+}
```

Comment:
I decided to not call this `toOpenPaymentsNonInteractiveGrant`, as I _think_ I like having just `Grant` or `InteractiveGrant`, thoughts? 

Depending on the decision, I will also update the exported types out of `open-payments`